onadata/libs/utils/csv_import.py

```diff
@@ -470,3 +366,239 @@ def submission_xls_to_csv(xls_file):
+                'Expected utf-8 encoded file or unicode'
+                ' string got {} instead'.format(type(csv_file).__name__)
+                ),
+            'valid': False}
+
+    # Change stream position to start of file
+    csv_file.seek(0)
+
+    csv_reader = ucsv.DictReader(csv_file, encoding='utf-8-sig')
+    csv_headers = csv_reader.fieldnames
+
+    # Make sure CSV headers have no spaces
+    # because these are converted to XLSForm names
+    # which cannot have spaces
+    if any(' ' in header for header in csv_headers):
+        return {
+            'error_msg': 'CSV file fieldnames should not contain spaces',
+            'valid': False}
+
+    # Get headers from stored data dictionary
+    xform_headers = xform.get_headers()
+
+    # Identify any missing columns between XForm and
+    # imported CSV ignoring repeat and metadata columns
+    missing_col = sorted([
+        col for col in set(xform_headers).difference(csv_headers)
+        if col.find('[') == -1 and not col.startswith('_')
+        and col not in IGNORED_COLUMNS and '/_' not in col])
+
+    mutliple_select_col = []
+
+    # Find all multiple_select type columns and remove them from
+    # the missing_col list
+    for col in csv_headers:
+        survey_element = xform.get_survey_element(col)
+        if survey_element and \
+                survey_element.get('type') == MULTIPLE_SELECT_TYPE:
+            # remove from the missing list
+            missing_col = [x for x in missing_col if not x.startswith(col)]
+            mutliple_select_col.append(col)
+
+    if missing_col:
+        return {
+            'error_msg': (
+                u"Sorry uploaded file does not match the form. "
+                u"The file is missing the column(s): "
+                u"{0}.".format(', '.join(missing_col))
+            ),
+            'valid': False}
+
+    # Identify any additional columns between XForm and
+    # imported CSV ignoring repeat and multiple_select columns
+    additional_col = [
+        col for col in set(csv_headers).difference(xform_headers)
+        if col.find('[') == -1 and col not in mutliple_select_col]
+
+    x_json = json.loads(xform.json)
+
+    # Retrieve the columns we should validate values for
+    # Currently validating date, datetime, integer and decimal columns
+    columns = {
+        'date': (get_columns_by_type(XLS_DATE_FIELDS, x_json), parse),
+        'datetime': (get_columns_by_type(XLS_DATETIME_FIELDS, x_json), parse),
+        'integer': (get_columns_by_type(['integer'], x_json), int),
+        'decimal': (get_columns_by_type(['decimal'], x_json), float)
+    }
+
+    validated_rows = []
+    errors = {}
+
+    for row_no, row in enumerate(csv_reader):
+        # Remove additional columns
+        for index in additional_col:
+            del row[index]
+
+        # Remove 'n/a' values and '' values from csv
+        row = {k: v for (k, v) in row.items() if v not in ['n/a', '']}
```

Comment:
use tag constant for `n/a` string, https://github.com/onaio/onadata/blob/7e0581d8475e61fd7ab911cf9940fc0a2845ecd6/onadata/libs/utils/common_tags.py#L67

Reply:
Changed.